src/cpu/kvm/vm.cc

```diff
@@ -457,23 +458,61 @@ KvmVM::setUserMemoryRegion(uint32_t slot,
                            void *host_addr, Addr guest_addr,
                            uint64_t len, uint32_t flags)
 {
-    struct kvm_userspace_memory_region m;
+    const long page_sysconf = sysconf(_SC_PAGESIZE);
+
+    // 4096 is the most common page size
+    const size_t page_size = (page_sysconf > 0) ? (size_t)page_sysconf : 4096;
```

Comment:
Nit: It may be a good idea to add the following here. I'll let you make the call.
```suggestion
    warn_if(page_sysconf <= 0, "Defaulting to 4 KiB page size for KVM alignment");
    const size_t page_size = (page_sysconf > 0) ? (size_t)page_sysconf : 4096;
```

Reply:
I agree, it's probably best to give users a hint in case sysconf fails and they have a different page size. I would hate to trying to debug that without any hints.